fly/src/syntax.rs

```diff
@@ -105,75 +162,314 @@ pub enum Term {
-    pub fn nary(op: NOp, lhs: Term, rhs: Term) -> Self {
-        Self::NAryOp(op, vec![lhs, rhs]).flatten_nary()
+    /// Smart constructor for prime. Note this does not push primes (which would
+    /// require a signature and context)
+    pub fn prime<T>(t: T) -> Self
+    where
+        T: Into<Term>,
+    {
+        Self::UnaryOp(UOp::Prime, Box::new(t.into()))
+    }
+
+    /// Smart constructor for always
+    pub fn always<T>(t: T) -> Self
+    where
+        T: Into<Term>,
+    {
+        Self::UnaryOp(UOp::Always, Box::new(t.into()))
+    }
+
+    /// Smart constructor for eventually
+    pub fn eventually<T>(t: T) -> Self
+    where
+        T: Into<Term>,
+    {
+        Self::UnaryOp(UOp::Eventually, Box::new(t.into()))
+    }
+
+    /// Smart constructor for next
+    pub fn next<T>(t: T) -> Self
+    where
+        T: Into<Term>,
+    {
+        Self::UnaryOp(UOp::Next, Box::new(t.into()))
+    }
+
+    /// Smart constructor for previous
+    pub fn previous<T>(t: T) -> Self
+    where
+        T: Into<Term>,
+    {
+        Self::UnaryOp(UOp::Previous, Box::new(t.into()))
+    }
+
+    //////////////////
+    // Binary operations: Equals, NotEquals, Implies, Iff, Until, Since
+    //////////////////
+
+    /// Smart constructor for `lhs = rhs`
+    pub fn equals<T1, T2>(lhs: T1, rhs: T2) -> Self
+    where
+        T1: Into<Term>,
+        T2: Into<Term>,
+    {
+        Self::BinOp(BinOp::Equals, Box::new(lhs.into()), Box::new(rhs.into()))
+    }
+
+    /// Smart constructor for `lhs != rhs`
+    pub fn not_equals<T1, T2>(lhs: T1, rhs: T2) -> Self
+    where
+        T1: Into<Term>,
+        T2: Into<Term>,
+    {
+        Self::BinOp(BinOp::NotEquals, Box::new(lhs.into()), Box::new(rhs.into()))
     }
 
-    /// Smart constructor equivalent to the And of an iterator of terms
+    /// Smart constructor for `lhs -> rhs`
+    pub fn implies<T1, T2>(lhs: T1, rhs: T2) -> Self
+    where
+        T1: Into<Term>,
+        T2: Into<Term>,
+    {
+        Self::BinOp(BinOp::Implies, Box::new(lhs.into()), Box::new(rhs.into()))
+    }
+
+    /// Smart constructor for `lhs <-> rhs`
+    pub fn iff<T1, T2>(lhs: T1, rhs: T2) -> Self
+    where
+        T1: Into<Term>,
+        T2: Into<Term>,
+    {
+        Self::BinOp(BinOp::Iff, Box::new(lhs.into()), Box::new(rhs.into()))
+    }
+
+    /// Smart constructor for `lhs until rhs`
+    pub fn until<T1, T2>(lhs: T1, rhs: T2) -> Self
+    where
+        T1: Into<Term>,
+        T2: Into<Term>,
+    {
+        Self::BinOp(BinOp::Until, Box::new(lhs.into()), Box::new(rhs.into()))
+    }
+
+    /// Smart constructor for `lhs since rhs`
+    pub fn since<T1, T2>(lhs: T1, rhs: T2) -> Self
+    where
+        T1: Into<Term>,
+        T2: Into<Term>,
+    {
+        Self::BinOp(BinOp::Since, Box::new(lhs.into()), Box::new(rhs.into()))
+    }
+
+    //////////////////
+    // N-ary operations: And, Or
+    //////////////////
+
+    /// Helper function for [`and`] and [`or`]
```

Comment:
```suggestion
    /// Helper function for [`Self::and`] and [`Self::or`]
```
(you actually have to fix this because it's a rustdoc warning we turned into an error)